pkg/gitutils/manifests.go

```diff
@@ -3,14 +3,38 @@ package gitutils
 import (
 	"embed"
 	"fmt"
-	"github.com/arlonproj/arlon/pkg/log"
-	gogit "github.com/go-git/go-git/v5"
 	"io"
+	"io/ioutil"
+	"os"
 	"path"
 	"strings"
+	"text/template"
+
+	"github.com/arlonproj/arlon/pkg/log"
+	"github.com/go-git/go-billy"
+	gogit "github.com/go-git/go-git/v5"
+	"gopkg.in/yaml.v2"
 )
 
-// -----------------------------------------------------------------------------
+// // -----------------------------------------------------------------------------
+
+type kustomizeyaml struct {
+	APIVersion     string   `yaml:"apiVersion"`
+	Kind           string   `yaml:"kind"`
+	Resources      []string `yaml:"resources"`
+	Configurations []string `yaml:"configurations"`
+	Patches        []target `yaml:"patches"`
+}
+type target struct {
+	Target info   `yaml:"target"`
+	Path   string `yaml:"path"`
+}
+
+type info struct {
```

Comment:
There's already a data GVK type in the controller-runtime, client-go, or Kubernetes library. Look for it and use it to avoid duplication. And why is this also duplicated in pkg/cluster/git.go above?

Reply:
Since, we are reading the values from a variable of type `map[interface{}]interface{}` the command `info.Object.GetObjectKind().GroupVersionKind()` is not able to work on it. And I removed the duplication of this struct in pkg/cluster/git.go file